Extensions/xkit_patches.js

```diff
@@ -384,9 +360,12 @@ XKit.extensions.xkit_patches = new Object({
 				}
 			};
 
-			XKit.tools.Nx_XHR = function(details) {
+			XKit.tools.Nx_XHR = details => new Promise((resolve, reject) => {
 				details.timestamp = new Date().getTime() + Math.random();
 
+				if (details.onload) { resolve = details.onload; }
```

Comment:
we should chain here instead of overwriting, so that the promise still works even if you pass in onload. (even though it's a legacy API, there might be intermediate functions where it makes sense to promisify some but not all requests)

Reply:
i feel like chaining them is more dangerous? since responses would get handled twice if you use both methods

Reply:
but then again whos gonna do that. okay